var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -68,20 +68,22 @@ class Amrex(CMakePackage, CudaPackage, ROCmPackage):
             description='Build particle classes')
     variant('plotfile_tools', default=False,
             description='Build plotfile_tools like fcompare')
-    variant('sundials', default=False,
-            description='Build AMReX with SUNDIALS support')
     variant('hdf5',  default=False,
             description='Enable HDF5-based I/O')
     variant('hypre', default=False,
             description='Enable Hypre interfaces')
     variant('petsc', default=False,
             description='Enable PETSc interfaces')
+    variant('sundials', default=False,
+            description='Enable SUNDIALS interfaces')
     variant('pic', default=False,
             description='Enable PIC')
 
     # Build dependencies
     depends_on('mpi', when='+mpi')
-    depends_on('sundials@4.0.0:4.1.0 +ARKODE +CVODE', when='@19.08: +sundials')
+    depends_on('sundials@4.0.0:4.1.0 +ARKODE +CVODE', when='@19.08:20.11 +sundials')
+    depends_on('sundials@5.7.0: +ARKODE +CVODE', when='@21.07: +sundials')
+    depends_on('sundials@5.7.0: +ARKODE +CVODE +cuda', when='@21.07: +sundials +cuda')
```

Comment:
```suggestion
    for arch in CudaPackage.cuda_arch_values:
        depends_on('sundials@5.7.0: +ARKODE +CVODE +cuda cuda_arch=%s'%arch, when='@21.07: +sundials +cuda cuda_arch=%s'%arch)
```

Reply:
@sayerhs's suggestion works.  I can install with `spack install amrex +cuda +sundials cuda_arch=xx` without extra `^sundials`.

Reply:
@balos1  Let's pass the amerx cuda_arch to sundials.

Reply:
@WeiqunZhang done.